### PR TITLE
Re-Run Jobs Validation

### LINKED ISSUE #3: Form Validation - Re-run past jobs
Currently a user can enter a letter half way through a number e.g `2222-22a40` and if the run is submitted an error is thrown with a stack trace.
Fix this so a user doesn't see the stack trace and instead friendly error saying they cannot enter letters in the 'Run Numbers' field.

Origin: https://github.com/mantidproject/autoreduce/issues/350.

### PR DESCRIPTION
**Pull request by DavidFair**
**Description of changes**

- Refactors converting string representations of a range of numbers into a separate file
- Adds new validation and better error messages when we do parse user output
- Incorrect input now returns an error to the user
- Run description and number of runs are limited by the server
- Travis CI Tests have been enabled for these changes
- Adds new run data to the populate DB SQL, so we can test multiple runs

- Minor: Removed unused imports where noticed

**How to test**
- Click re-run job:
- Enter a nonsense input such as 100-11a0
- For MUSR try any range of runs between 1 and 10, check it submits correctly
- Try an input with more than 500 runs (for admins), or less than 50 (for staff)
- Comment out the following lines in `run_variables.js` to disable client side checking:
```
            var max_length = 200;
            if($(this).val().length >= max_length) {
                addInvalid($(this), 'Re-run description must be less than ' + max_length.toString() + ' characters.')
            }
```
- Then try to enter a description more than 200 characters

Fixes #3 and #4 

---

**To test**
* Log into the devolpement nodes
* `git checkout this-pull-request-branch`
* restart services on each node (information on how to do this can be found in the development documentation)
* Submit some runs with the manualsubmission.py script

Once you are happy, merge this pull request into develop, and update the development nodes to the devlop branch
```
> git checkout origin/devlop
> git fetch -p
> git pull origin/develop
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -40,3 +40,4 @@ script:
 
   # Utils tests
   - nosetests utils/clients/tests
+  - nosetests WebApp/autoreduce_webapp/test
```
Comment (ElliotAOram) [inline .travis.yml]: Should this dir be `tests` rather than `test`?

File: WebApp/autoreduce_webapp/test/oldtest_reduction_variables.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import logging, os, sys, shutil, imp, json
+import logging, os, sys, json
```
Comment (ElliotAOram) [inline WebApp/autoreduce_webapp/test/oldtest_reduction_variables.py]: Might be worth commenting at the top of the old test classes to say that they need to be assessed for validity/usefulness - just for clarity.

**Comment by DavidFair** [comment]
This branch is ready to be reviewed/merged

**Comment by ElliotAOram** [comment]
@DavidFair if I enter negative numbers into the runNumbers box on the Re-run page and press submit I get a django error.
image
image

**Comment by ElliotAOram** [comment]
There is inconsistency between the way invalid inputs are given back to the user e.g.
`a23-465b` will take me to a whole new page stating that the character is invalid.
`a23,45b` will keep be on the same page and tell me the input is invalid.

My guess is that it is attempting to expand the dash before validating the characters?

**Comment by DavidFair** [comment]
@ElliotAOram This should be ready for testing again

**Pull request merged**